tests/e2e_logging.py

```diff
@@ -72,6 +72,39 @@ def test_cert_prefix(network, args):
     return network
 
 
+@reqs.description("Write as anonymous caller")
+@reqs.supports_methods("LOG_record_prefix_cert", "LOG_get")
+def test_anonymous_caller(network, args):
+    if args.package == "liblogging":
+        primary, _ = network.find_primary()
+
+        # Create a new user but do not record its identity
+        network.create_user(4, args.default_curve)
+
+        log_id = 101
+        msg = "This message is anonymous"
+        with primary.user_client(user_id=4) as c:
+            r = c.rpc("LOG_record_anonymous", {"id": log_id, "msg": msg})
+            assert r.result == True
+            r = c.rpc("LOG_get", {"id": log_id})
+            assert (
+                r.error is not None
+            ), "Anonymous user is not authorised to call LOG_get"
+
+        with primary.user_client(user_id=0) as c:
+            r = c.rpc("LOG_record_anonymous", {"id": log_id, "msg": msg})
+            assert (
+                r.error is not None
+            ), "Only anonymous users are authorised to call LOG_record_anonymous"
+            r = c.rpc("LOG_get", {"id": log_id})
+            assert r.result is not None
+            assert f"Anonymous: {msg}" in r.result["msg"]
```

Comment:
Minor: I think this should just be `assert msg in r.result["msg"]` - we want to check that the anonymous message was recorded, and the fact its prefixed (and precisely _how_ its prefixed) seem uninteresting.